### PR TITLE
feat(performance): Add Queue module chart components

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds latency and throughput charts with tests for the queues module

### CODE REVIEW

File: static/app/views/performance/queues/charts/latencyChart.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          yAxis: [
+            'avg_if(span.self_time,span.op,queue.submit.celery)',
+            'avg_if(span.self_time,span.op,queue.task.celery)',
+            'count_op(queue.submit.celery)',
+            'count_op(queue.task.celery)',
+          ],
+        }),
+      })
+    );
+  });
```
Comment (gggritso) [inline static/app/views/performance/queues/charts/latencyChart.spec.tsx:41]: I _think_ it's a good idea here, if possible, to `await` when the loading spinner disappears, so the test can unmount cleanly, but worth asking the FE channel if that's important or not

File: static/app/views/performance/queues/charts/latencyChart.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export function LatencyChart({error}: Props) {
+  const {query} = useLocation();
+  const destination = decodeScalar(query.destination);
+  const pageFilters = usePageFilters();
+  const period = pageFilters.selection.datetime.period;
+  const chartSubtext = (period && DEFAULT_RELATIVE_PERIODS[period]) ?? '';
+  const {data, isLoading} = useQueuesTimeSeriesQuery({destination});
+  return (
+    <ChartPanel title={t('Avg Latency')}>
+      <ChartSubtext>{chartSubtext}</ChartSubtext>
```
Comment (gggritso) [inline static/app/views/performance/queues/charts/latencyChart.tsx:28]: `ChartPanel` supports a `subtitle` prop, does that help in this case?

File: static/app/views/performance/queues/charts/latencyChart.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        grid={{
+          left: '0',
+          right: '0',
+          top: '8px',
+          bottom: '0',
+        }}
+        data={
+          [
+            {
+              seriesName: 'Average Processing Latency',
```
Comment (gggritso) [inline static/app/views/performance/queues/charts/latencyChart.tsx:40]:
```suggestion
              seriesName: t('Average Processing Latency'),
```

File: static/app/views/performance/queues/charts/throughputChart.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        grid={{
+          left: '0',
+          right: '0',
+          top: '8px',
+          bottom: '0',
+        }}
+        data={
+          [
+            {
+              seriesName: 'Published',
```
Comment (gggritso) [inline static/app/views/performance/queues/charts/throughputChart.tsx:40]:
```suggestion
              seriesName: t('Published'),
```

File: static/app/views/performance/queues/charts/throughputChart.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+              data: data['count_op(queue.submit.celery)'],
+            },
+            {
+              seriesName: 'Processed',
```
Comment (gggritso) [inline static/app/views/performance/queues/charts/throughputChart.tsx:44]:
```suggestion
              seriesName: t('Processed'),
```

**Review by gggritso — APPROVED** [review]
Makes sense! Watch out for custom `<ChartSubtext>` which might not be necessary, and awaiting the full loaded state of a chart

File: static/app/views/performance/queues/charts/latencyChart.spec.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/charts/latencyChart.spec.tsx]: updated to await for loading spinners!

File: static/app/views/performance/queues/charts/latencyChart.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/charts/latencyChart.tsx]: 👍 good call, updated to use subtitle

**Pull request merged**